plugins/providers/alicloudiam/client.go

```diff
@@ -0,0 +1,332 @@
+package alicloudiam
+
+import (
+	"context"
+	"fmt"
+	openapi "github.com/alibabacloud-go/darabonba-openapi/v2/client"
+	ram "github.com/alibabacloud-go/ram-20150501/v2/client"
+	utils "github.com/alibabacloud-go/tea-utils/v2/service"
+	"github.com/aliyun/credentials-go/credentials"
+	"github.com/goto/guardian/domain"
+	"golang.org/x/sync/errgroup"
+	"strings"
+	"sync"
+)
+
+const (
+	PolicyTypeSystem = "System"
+	PolicyTypeCustom = "Custom"
+
+	aliAccountUserIdDomainSuffix = ".onaliyun.com"
+	aliAccountUserIdPattern      = `^[a-zA-Z0-9._%+-]+@[0-9]+\.onaliyun\.com$`
+)
+
+type iamClient struct {
+	resourceName string
+	iamService   *ram.Client
+}
+
+func newIamClient(accessKeyID, accessKeySecret, resourceName string) (*iamClient, error) {
+	credsConfig := new(credentials.Config).
+		SetType("access_key").
+		SetAccessKeyId(accessKeyID).
```

Comment:
The accessKeyID and accessKeySecret will be received in base64 so that it is not exposed. So you can make the changes to decrypt and then use the values.

Reply:
Before implementing the changes, I have a question Ms. If we encode the access key on base 64. On the API request body it will become like this when creating a alicloudiam provider:
```
{
    "type": "alicloud_iam",
    "urn": "55890119897629982",
    "allowed_account_types": [
        "ramUser",
        "ramRole"
    ],
    "credentials": {
        "access_key_id": "Original AccessKeyID", -> "Encoded AccessKeyID"
        "access_key_secret": "Original AccessKeySecret", -> "Encoded AccessKeySecret"
        "resource_name": "accounts/5589011989762998"
    },
    "appeal": {
        "allow_permanent_access": false,
        "allow_active_access_extension_in": "336h"
    },
    "resources": [
        {
            "type": "account",
            "policy": {
                "id": "alicloud_account_policy",
                "version": 1
            },
            "roles": [
                {
                    "id": "AliyunOSSReadOnlyAccess",
                    "name": "AliyunOSSReadOnlyAccess",
                    "description": "",
                    "permissions": [
                        "AliyunOSSReadOnlyAccess"
                    ]
                }
            ]
        }
    ]
}
```

So when we retrieve the access key from alicloud, we need to manually to convert it to base64 first for both id and secret.

As far as I know, when you create a new service account on Google. We received default encoded secret.
But, on alicloud we received an plain secret.

If you are okay with this, I will implement the changes.

Reply:
But for the encryption & decryption, I have already implemented on this function:
```func (p *Provider) CreateConfig(pc *domain.ProviderConfig) error ```

So it will encrypt the secret immediately when the core package received a command to create an provider.
And it able to decrypt again when they are needed. Is this implementation already ok Ms? :D

Reply:
So when we retrieve the access key from alicloud, we need to manually to convert it to base64 first for both id and secret. Then put it on the create provider request body.
Yes, this is correct. We will provide the encrypted value in the provider request body

Reply:
Got it, ack!

Reply:
Resolved: https://github.com/goto/guardian/pull/187/commits/811bf152dfd0de9f2e115a122bb642207bb4e9b1